service/frontend/wrappers/ratelimited/ratelimit.go

```diff
@@ -0,0 +1,49 @@
+// The MIT License (MIT)
+
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package ratelimited
+
+import (
+	"github.com/uber/cadence/common/quotas"
+)
+
+// ratelimitType differentiates between the three categories of ratelimiters
+type ratelimitType int
+
+const (
+	ratelimitTypeUser ratelimitType = iota + 1
+	ratelimitTypeWorker
+	ratelimitTypeVisibility
+)
+
+func (h *apiHandler) allowDomain(requestType ratelimitType, domain string) bool {
+	switch requestType {
+	case ratelimitTypeUser:
+		return h.userRateLimiter.Allow(quotas.Info{Domain: domain})
+	case ratelimitTypeWorker:
+		return h.workerRateLimiter.Allow(quotas.Info{Domain: domain})
+	case ratelimitTypeVisibility:
+		return h.visibilityRateLimiter.Allow(quotas.Info{Domain: domain})
+	default:
+		panic("coding error, unrecognized request ratelimit type value")
+	}
+}
```

Comment:
this is a fairly minimal change and if you prefer that then 👍 but:
we could take this opportunity to remove this method and directly call the appropriate limiter.

I didn't do that initially when adding these, to keep things minimal, but it's definitely due for a cleanup at some point.  Does not have to be now though.